% Sweave(file = './deletion-tdt.Rnw')
\documentclass[10pt, final]{article}
\usepackage{Sweave}
\SweaveOpts{eps=FALSE,echo=TRUE,figs.only=TRUE,keep.source=FALSE, prefix.string=figures/deletion-tdt}
%\usepackage{fullpage}
%\usepackage{times}
%\usepackage[colorlinks=TRUE,urlcolor=blue,citecolor=blue]{hyperref}
%\usepackage{ifthen}  % Conditional 
%\usepackage{multicol}   %Columns
\usepackage{graphicx} %should be removed before submission!
\input{/home/sgy/jhsph/latex/sgy}
\begin{document}
<<options, echo=FALSE, eval = TRUE, echo = FALSE>>=
  options(width=75, continue = " ")
  library("Gviz")
  library("trioClasses")
  library("TxDb.Hsapiens.UCSC.hg18.knownGene")
  data("fe", package = "trioClasses")
  data("pedigrees", package="CleftCNVAssoc")
  data("penncnvjoint", package = "CleftCNVAssoc")
  data("cnv", package = "trioClasses")
@ 

<<FamilyExperiment, echo = FALSE, eval = TRUE>>=
  fe.beaty.parents <- fe.beaty[,colnames(fe.beaty)%in%parents(fe.beaty)]
  fe.pitt.parents <- fe.pitt[,colnames(fe.pitt)%in%parents(fe.pitt)]
@ 
<<freq-vec, echo = FALSE, eval = TRUE>>=
    freq.beaty.vec <- colSums(cnv(fe.beaty.parents))/nrow(cnv(fe.beaty.parents))
    freq.pitt.vec <- colSums(cnv(fe.pitt.parents))/nrow(cnv(fe.pitt.parents))
@ 
<<trioStates, eval = TRUE, echo = FALSE>>=
    trioAssay.beaty <- trioClasses:::TrioAssay(fe.beaty, type = "cnv")
    trioStates.beaty <- with(trioAssay.beaty, matrix( paste0(F,M,O), nrow = nrow(O), ncol = ncol(O)))
    dimnames(trioStates.beaty) <- dimnames(trioAssay.beaty$O)
    trioAssay.pitt <- trioClasses:::TrioAssay(fe.pitt, type = "cnv")
    trioStates.pitt <- with(trioAssay.pitt, matrix( paste0(F,M,O), nrow = nrow(O), ncol = ncol(O)))
    dimnames(trioStates.pitt) <- dimnames(trioAssay.pitt$O)
@ 
<<table-list, eval = TRUE, echo = FALSE>>=
    table.list.beaty <- apply(trioStates.beaty, 2, "table")
    table.list.pitt <- apply(trioStates.pitt, 2, "table")
@ 
The top CNV component has the followinfg trio-states.  Where `1' indicates a deletion and order is F, M, O.
<<transtab, eval = TRUE, echo = FALSE>>=
    trans.vec <- as( lapply( table.list.beaty, trioClasses:::trans.tab ), "numeric")
    trans.vec.pitt <- as( lapply( table.list.pitt, trioClasses:::trans.tab ), "numeric")

    binom.list.beaty <- lapply( table.list.beaty, trioClasses:::trans.rate )
    binom.list.pitt <- lapply( table.list.pitt, trioClasses:::trans.rate )

@ 

<<ci-beaty, echo = FALSE>>=
est.list.beaty <- lapply(binom.list.beaty, FUN = trioClasses:::get.est)
ci.list.beaty <- lapply(binom.list.beaty, FUN = trioClasses:::get.ci)
testable.beaty <- !is.na(ci.list.beaty)
ci.mat.beaty <- matrix(unlist(ci.list.beaty[testable.beaty]), nrow = sum(testable.beaty), ncol = 2, byrow = TRUE)
est.vec.beaty <- as(est.list.beaty, "numeric")[testable.beaty]
gr.cnp.beaty <- cnv.obj.beaty$cmp.gr[testable.beaty]
freq.beaty.vec <- freq.beaty.vec[testable.beaty]
@ 

<<ci-pitt, echo = FALSE>>=
est.list.pitt <- lapply(binom.list.pitt, FUN = trioClasses:::get.est)
ci.list.pitt <- lapply(binom.list.pitt, FUN = trioClasses:::get.ci)
testable.pitt <- !is.na(ci.list.pitt)
ci.mat.pitt <- matrix(unlist(ci.list.pitt[testable.pitt]), nrow = sum(testable.pitt), ncol = 2, byrow = TRUE)
est.vec.pitt <- as(est.list.pitt, "numeric")[testable.pitt]
gr.cnp.pitt <- cnv.obj.pitt$cmp.gr[testable.pitt]
freq.pitt.vec <- freq.pitt.vec[testable.pitt]
@ 

\begin{center}
\begin{figure}
<<trans-ci, fig = TRUE, echo = FALSE, eval = TRUE, width = 6>>=
par(bg = '#FFFEDB')
pch <- substr(as.character(seqnames(cnv.obj.beaty$cmp.gr[testable.beaty])),4,999)
col <- as(pch,"integer")
plot( est.vec.beaty, pch = 20,col = col, axes = FALSE, ylim = c(0,1) )
lines(c(1,length(est.vec.beaty)), rep(0.5,2), lty = 3)
legend(x = 1, y = 1, legend = 1:22, col = 1:22, pch = 20)
axis(2)
axis(2)
@ 
\end{figure}
\end{center}
\begin{center}
\begin{figure}
<<trans-ci-pitt, fig = TRUE, echo = FALSE, eval = TRUE, width = 6>>=
par(bg = '#FFFEDB')
pch <- substr(as.character(seqnames(cnv.obj.pitt$cmp.gr[testable.pitt])),4,999)
col <- as(pch,"integer")
plot( est.vec.pitt, pch = 20,col = col, axes = FALSE, ylim = c(0,1) )
lines(c(1,length(est.vec.pitt)), rep(0.5,2), lty = 3)
legend(x = 1, y = 1, legend = 1:22, col = 1:22, pch = 20)
axis(2)
axis(2)
@ 
\end{figure}
\end{center}
%% \begin{center}
%% \begin{figure}
%% <<trans-ci, fig = TRUE, echo = FALSE, eval = TRUE, height = 12>>=
%% par(bg = '#FFFEDB')
%% plot(1, type = "n", ylim = c(0,max(length(est.vec.beaty),length(est.vec.pitt))), xlim = c(0,2), xlab = "Transmission Rate", ylab = "", main = "Transmission Rate 95 percent CI\nT+U>50 ", axes = FALSE )
%% polygon( y = c(1:length(est.vec.beaty), length(est.vec.beaty):1 ), x = c(ci.mat.beaty[,1],rev(ci.mat.beaty[,2])), col = "darkblue", border = NA)

%% polygon( y = c(1:length(est.vec.pitt), length(est.vec.pitt):1 ), x = c(ci.mat.pitt[,1],rev(ci.mat.pitt[,2]))+1, col = "orange", border = NA)
%% lines( x = rep(0.5,2), y = c(0,max(length(est.vec.beaty),length(est.vec.pitt))), lty = 2, lwd = 2)
%% lines( x = rep(0.5,2)+1, y = c(0,max(length(est.vec.beaty),length(est.vec.pitt))), lty = 2, lwd = 2)
%% lines( x = rep(1,2), y = c(0,max(length(est.vec.beaty),length(est.vec.pitt))), lty = 3, lwd = 2)
%% axis(1, at = c(0.5,1.5), labels = c(0.5,0.5) )
%% axis(2)
%% @ 
%% \end{figure}
%% \end{center}


\begin{center}
  \begin{figure}
    \includegraphics[width = 4in]{./figures/ucsc}
    \caption{UCSC Genome Browser at chromosome 8 locus.}
  \end{figure}
\end{center}
<<n-tests, eval = TRUE, echo = FALSE>>=
    n.tests <- sum(!is.na(trans.vec))
@ 
The CNV components with signiifcant $p$-values (Bonferroni) are given below.
<<reduce-locus, eval = TRUE, echo = FALSE >>=
    (locus <- cnv.obj.beaty$cmp.gr[which(trans.vec <= 0.05/n.tests)])
    wd <- width(reduce(locus))/1e3
@ 

<<locus-full, echo = FALSE>>=
(locus.full <- reduce(reduce(cnv.obj.beaty$cmp.gr)[queryHits(findOverlaps(reduce(cnv.obj.beaty$cmp.gr), locus)) ]))

index.1.beaty <- queryHits(findOverlaps(cnv.obj.beaty$cmp.gr, locus[1]))
index.1.full.beaty <- queryHits(findOverlaps(cnv.obj.beaty$cmp.gr, locus.full[1]))
index.2.beaty <- queryHits(findOverlaps(cnv.obj.beaty$cmp.gr, locus[2]))
index.3.beaty <- queryHits(findOverlaps(cnv.obj.beaty$cmp.gr, locus[3]))
index.2.full.beaty <- queryHits(findOverlaps(cnv.obj.beaty$cmp.gr, locus.full[2]))
index.4.beaty <- queryHits(findOverlaps(cnv.obj.beaty$cmp.gr, locus[4]))
index.3.full.beaty <- queryHits(findOverlaps(cnv.obj.beaty$cmp.gr, locus.full[3]))
index.1.pitt <- queryHits(findOverlaps(cnv.obj.pitt$cmp.gr, locus[1]))
index.1.full.pitt <- queryHits(findOverlaps(cnv.obj.pitt$cmp.gr, locus.full[1]))
index.2.pitt <- queryHits(findOverlaps(cnv.obj.pitt$cmp.gr, locus[2]))
index.3.pitt <- queryHits(findOverlaps(cnv.obj.pitt$cmp.gr, locus[3]))
index.2.full.pitt <- queryHits(findOverlaps(cnv.obj.pitt$cmp.gr, locus.full[2]))
index.4.pitt <- queryHits(findOverlaps(cnv.obj.pitt$cmp.gr, locus[4]))
index.3.full.pitt <- queryHits(findOverlaps(cnv.obj.pitt$cmp.gr, locus.full[3]))

@ 

<<freq, echo = TRUE>>=
freq.beaty <- colSums(cnv(fe.beaty))/2/nrow(cnv(fe.beaty))
freq.pitt <- colSums(cnv(fe.pitt))/2/nrow(cnv(fe.pitt))

freq.beaty[index.1.beaty]
freq.pitt[index.1.pitt]

freq.beaty[index.2.beaty]
freq.pitt[index.2.pitt]

freq.beaty[index.3.beaty]
freq.pitt[index.3.pitt]

freq.beaty[index.4.beaty]
freq.pitt[index.4.pitt]

summary(freq.beaty[index.1.full.beaty])
summary(freq.pitt[index.1.full.pitt])

summary(freq.beaty[index.2.full.beaty])
summary(freq.pitt[index.2.full.pitt])

summary(freq.beaty[index.3.full.beaty])
summary(freq.pitt[index.3.full.pitt])
@ 

<<trans.tab, eval = TRUE, echo = TRUE >>=
    trioClasses:::trans.tab
@ 
<<trans.rate, eval = TRUE, echo = TRUE >>=
    trioClasses:::trans.rate
@ 

\section*{Methods}
\subsection*{Cleft Data Description}
\begin{itemize}
  \item Performed \Sexpr{n.tests} tests.  %Bonferroni significant locus has width \Sexpr{wd} kB.
  \item \penn{} joint HMM
  \item european, MAD $<0.3$, non-WGA, aux $\ne 1$
  \item coverage $>10$
  \item \Sexpr{length(cnv.obj.beaty$gr)} hemi/homozygous deletions identified in \Sexpr{nrow(completeTrios(fe.beaty))} trios
  \item \Sexpr{ncol(cnv(fe.beaty))} CNV components
  \subitem Common ($>0.01$): \Sexpr{sum(freq.beaty > 0.01)}
  \subitem Rare: \Sexpr{sum(freq.beaty <= 0.01)}
  \item Construct trio-states for all CNV components
    \subitem 0,1,2 variable for normal, hemizygous and homozygous, respectively
  \item $T + U > 50$
  \item count transmissions and perform binom.test (See ``trans.tab'')
\end{itemize}


\begin{center}
\begin{figure}
<<hist, fig = TRUE, echo = FALSE, eval = TRUE, width = 8>>=
par(bg = '#FFFEDB')
layout(matrix(1:2, nrow = 1, ncol = 2))
hist(est.vec.beaty, breaks = 20, xlim = c(0,1), ylim = c(0,100), main = "Cleft Trios", xlab = "Transmission Rate", col='darkblue')
hist(est.vec.pitt, breaks = 20, xlim = c(0,1), ylim = c(0,100), main = "Control Trios", xlab = "Transmission Rate", col='darkblue')
@ 
\end{figure}
\end{center}

\begin{center}
\begin{figure}
<<gviz, eval = TRUE, echo = FALSE, fig = TRUE, width = 6>>=
chr <- 8
gr.cnp.beaty.chr <- gr.cnp.beaty[as.logical(seqnames(gr.cnp.beaty) == paste0("chr", chr))]

gr.cnp.pitt.chr <- gr.cnp.pitt[as.logical(seqnames(gr.cnp.pitt) == paste0("chr", chr))]

est.vec.beaty.chr <- est.vec.beaty[as.logical(seqnames(gr.cnp.beaty) == paste0("chr", chr))]
freq.vec.beaty.chr <- freq.beaty.vec[as.logical(seqnames(gr.cnp.beaty) == paste0("chr", chr))]
lower.vec.beaty.chr <- ci.mat.beaty[as.logical(seqnames(gr.cnp.beaty) == paste0("chr", chr)),1]
upper.vec.beaty.chr <- ci.mat.beaty[as.logical(seqnames(gr.cnp.beaty) == paste0("chr", chr)),2]

est.vec.pitt.chr <- est.vec.pitt[as.logical(seqnames(gr.cnp.pitt) == paste0("chr", chr))]
freq.vec.pitt.chr <- freq.pitt.vec[as.logical(seqnames(gr.cnp.pitt) == paste0("chr", chr))]
lower.vec.pitt.chr <- ci.mat.pitt[as.logical(seqnames(gr.cnp.pitt) == paste0("chr", chr)),1]
upper.vec.pitt.chr <- ci.mat.pitt[as.logical(seqnames(gr.cnp.pitt) == paste0("chr", chr)),2]


TranscriptDb <- TxDb.Hsapiens.UCSC.hg18.knownGene
atrack <- AnnotationTrack(, name = "CNP comp.", fill = "darkgreen")
gtrack <- GenomeAxisTrack()
itrack <- IdeogramTrack(genome = "hg18", chromosome = paste0("chr", chr), lty = 1, lwd = 1 )
grtrack <- GeneRegionTrack(TranscriptDb, genome="hg18", chromosome=chr, name="UCSC Known Genes", geneSymbols = TRUE, collapseTranscripts=TRUE, showId = FALSE)

dtrack <- DataTrack( range = gr.cnp.beaty.chr, data = rbind(est.vec.beaty.chr,  lower.vec.beaty.chr, upper.vec.beaty.chr), type = "S", col = c("orange", "darkblue","orange"), lwd = 2, name = "Cleft Trios", ylim = c(0.25,0.75), pch = 20 )
dtrack2 <- DataTrack( range = gr.cnp.pitt.chr, data = rbind(est.vec.pitt.chr,  lower.vec.pitt.chr, upper.vec.pitt.chr), type = "S", , col = c("orange", "darkblue","orange"), lwd = 2, name = "Control Trios", ylim = c(0.25,0.75), pch = 20 )
fud <- 4e5/2
plotTracks( list(dtrack,  dtrack2, grtrack, gtrack, itrack ),  groups = c("trans","lower","upper"), background.panel = '#FFFEDB', background.title = 'darkblue', from = mean(c(39356825,39497557)) - fud, to = mean(c(39356825,39497557)) + fud, legend = FALSE)

@ 
\label{chr8-locus}
\caption{}
\end{figure}
\end{center}




\begin{center}
\begin{figure}
<<gviz2, eval = TRUE, echo = FALSE, fig = TRUE, width = 6>>=
chr <- 7

gr.cnp.beaty.chr <- gr.cnp.beaty[as.logical(seqnames(gr.cnp.beaty) == paste0("chr", chr))]


gr.cnp.pitt.chr <- gr.cnp.pitt[as.logical(seqnames(gr.cnp.pitt) == paste0("chr", chr))]

est.vec.beaty.chr <- est.vec.beaty[as.logical(seqnames(gr.cnp.beaty) == paste0("chr", chr))]
freq.vec.beaty.chr <- freq.beaty.vec[as.logical(seqnames(gr.cnp.beaty) == paste0("chr", chr))]
lower.vec.beaty.chr <- ci.mat.beaty[as.logical(seqnames(gr.cnp.beaty) == paste0("chr", chr)),1]
upper.vec.beaty.chr <- ci.mat.beaty[as.logical(seqnames(gr.cnp.beaty) == paste0("chr", chr)),2]

est.vec.pitt.chr <- est.vec.pitt[as.logical(seqnames(gr.cnp.pitt) == paste0("chr", chr))]
freq.vec.pitt.chr <- freq.pitt.vec[as.logical(seqnames(gr.cnp.pitt) == paste0("chr", chr))]
lower.vec.pitt.chr <- ci.mat.pitt[as.logical(seqnames(gr.cnp.pitt) == paste0("chr", chr)),1]
upper.vec.pitt.chr <- ci.mat.pitt[as.logical(seqnames(gr.cnp.pitt) == paste0("chr", chr)),2]


atrack <- AnnotationTrack(, name = "CNP comp.", fill = "darkgreen")
gtrack <- GenomeAxisTrack()
itrack <- IdeogramTrack(genome = "hg18", chromosome = paste0("chr", chr), lty = 1, lwd = 1 )
grtrack <- GeneRegionTrack(TranscriptDb, genome="hg18", chromosome=chr, name="UCSC Known Genes", geneSymbols = TRUE, collapseTranscripts=TRUE, showId = FALSE)

dtrack <- DataTrack( range = gr.cnp.beaty.chr, data = rbind(est.vec.beaty.chr,  lower.vec.beaty.chr, upper.vec.beaty.chr), type = "S", col = c("orange", "darkblue","orange"), lwd = 2, name = "Cleft Trios", ylim = c(0,1), pch = 20 )
dtrack2 <- DataTrack( range = gr.cnp.pitt.chr, data = rbind(est.vec.pitt.chr,  lower.vec.pitt.chr, upper.vec.pitt.chr), type = "S", , col = c("orange", "darkblue","orange"), lwd = 2, name = "Control Trios", ylim = c(0,1), pch = 20 )

#fud <- 3e5/2

plotTracks( list(dtrack, dtrack2, grtrack, gtrack, itrack  ),  groups = c("trans","lower","upper"), background.panel = '#FFFEDB', background.title = 'darkblue', from = 141.43e6 - fud, to = 141.43e6 + fud, legend = FALSE)

@ 
\label{chr7-locus}
\caption{}
\end{figure}
\end{center}


\begin{center}
\begin{figure}
<<gviz3, eval = TRUE, echo = FALSE, fig = TRUE, width = 6>>=
chr <- 11

gr.cnp.beaty.chr <- gr.cnp.beaty[as.logical(seqnames(gr.cnp.beaty) == paste0("chr", chr))]


gr.cnp.pitt.chr <- gr.cnp.pitt[as.logical(seqnames(gr.cnp.pitt) == paste0("chr", chr))]


est.vec.beaty.chr <- est.vec.beaty[as.logical(seqnames(gr.cnp.beaty) == paste0("chr", chr))]
freq.vec.beaty.chr <- freq.beaty.vec[as.logical(seqnames(gr.cnp.beaty) == paste0("chr", chr))]
lower.vec.beaty.chr <- ci.mat.beaty[as.logical(seqnames(gr.cnp.beaty) == paste0("chr", chr)),1]
upper.vec.beaty.chr <- ci.mat.beaty[as.logical(seqnames(gr.cnp.beaty) == paste0("chr", chr)),2]

est.vec.pitt.chr <- est.vec.pitt[as.logical(seqnames(gr.cnp.pitt) == paste0("chr", chr))]
freq.vec.pitt.chr <- freq.pitt.vec[as.logical(seqnames(gr.cnp.pitt) == paste0("chr", chr))]
lower.vec.pitt.chr <- ci.mat.pitt[as.logical(seqnames(gr.cnp.pitt) == paste0("chr", chr)),1]
upper.vec.pitt.chr <- ci.mat.pitt[as.logical(seqnames(gr.cnp.pitt) == paste0("chr", chr)),2]


atrack <- AnnotationTrack(, name = "CNP comp.", fill = "darkgreen")
gtrack <- GenomeAxisTrack()
itrack <- IdeogramTrack(genome = "hg18", chromosome = paste0("chr", chr), lty = 1, lwd = 1 )
grtrack <- GeneRegionTrack(TranscriptDb, genome="hg18", chromosome=chr, name="UCSC Known Genes", geneSymbols = TRUE, collapseTranscripts=TRUE, showId = FALSE)

dtrack <- DataTrack( range = gr.cnp.beaty.chr, data = rbind(est.vec.beaty.chr,  lower.vec.beaty.chr, upper.vec.beaty.chr), type = "S", col = c("orange", "darkblue","orange"), lwd = 2, name = "Cleft Trios", ylim = c(0,1), pch = 20 )
dtrack2 <- DataTrack( range = gr.cnp.pitt.chr, data = rbind(est.vec.pitt.chr,  lower.vec.pitt.chr, upper.vec.pitt.chr), type = "S", , col = c("orange", "darkblue","orange"), lwd = 2, name = "Control Trios", ylim = c(0,1), pch = 20 )

#fud <- 3e5

plotTracks( list(dtrack, dtrack2, grtrack, gtrack, itrack  ),  groups = c("trans","lower","upper"), background.panel = '#FFFEDB', background.title = 'darkblue', from = 55.20e6 - fud, to = 55.20e6 + fud, legend = FALSE)

@ 
\label{chr11-locus}
\caption{}
\end{figure}
\end{center}

\end{document}
