% Sweave(file = "./deletion-tdt.Rnw")
\documentclass[10pt, final]{bmc_article}
<<echo=FALSE>>=
  options(width=75, continue = " ")
@
\SweaveOpts{eps=FALSE,echo=TRUE,figs.only=TRUE,keep.source=FALSE, prefix.string=figures/deletion-tdt}
\usepackage{fullpage}
\usepackage{times}
\usepackage[colorlinks=TRUE,urlcolor=blue,citecolor=blue]{hyperref}
\usepackage{ifthen}  % Conditional 
\usepackage{multicol}   %Columns
\usepackage{graphicx} %should be removed before submission!
\input{/home/sgy/jhsph/latex/sgy}

\newboolean{publ}
%Review style settings
\newenvironment{bmcformat}{\begin{raggedright}\baselineskip20pt\sloppy\setboolean{publ}{false}}{\end{raggedright}\baselineskip20pt\sloppy}
%Publication style settings
%\newenvironment{bmcformat}{\fussy\setboolean{publ}{true}}{\fussy}

\begin{document}
\begin{bmcformat}

\title{Inherited deletions!}
\author{Samuel G Younkin$^1$\and Robert B Scharpf$^2$\and Holger Schwender$^3$\and Alan F Scott$^4$\and Mary L Marazita$^5$\and Terri H Beaty$^6$ and Ingo Ruczinski\correspondingauthor$^1$\email{ingo@jhu.edu}}

\address{
  \iid(1) Department of Biostatistics, Johns Hopkins Bloomberg School of Public Health, Baltimore, MD, USA; 
  \iid(2) Department of Oncology, Johns Hopkins School of Medicine, Baltimore, MD, USA;
  \iid(3) Mathematical Institute, Heinrich-Heine-University, D\"usseldorf, Germany;
  \iid(4) Department of Medicine, Johns Hopkins School of Medicine, Baltimore, MD, USA;
  \iid(5) School of Dental Medicine, University of Pittsburgh, Pittsburgh, PA, USA;
  \iid(6) Department of Epidemiology, Johns Hopkins Bloomberg School of Public Health, Baltimore, MD, USA
}

\maketitle

\begin{abstract}
  
  Copy number variants (CNVs) may play an important part in the
  development of common birth defects such as oral clefts, and
  individual patients with multiple birth defects (including clefts)
  have been shown to carry small and large chromosomal deletions.  In
  this paper we investigate the transmission rates of deletions in
  cleft offspring-parent trios.
    
  %%   We compare de-novo deletion frequencies in Caucasian
  %% children with cleft lip with or without cleft palate
  %% (CL/P) to frequencies in Caucasian children from triads
  %% not ascertained for a specific phenotype.

  %% We identified a genome-wide significant 62 kB non-coding
  %% region on chromosome 7p14.1 where de-novo deletions occur
  %% more frequently in oral cleft cases than controls.
  
  %% We also observed an increase in the width of de-novo
  %% deletions in oral cleft probands, with cleft lip with or
  %% without cleft palate (CL/P) cases exhibiting wider de-novo
  %% deletions than cleft palate (CP) and cleft lip (CL) cases.

  %% De-novo deletions appear to be involved in the etiology of
  %% \clp, although the underlying biological mechanism are
  %% still unknown.
  
  %% Larger de-novo deletions are more likely to interfere with
  %% regular development and may result in oral clefting.
  
  %% Study protocol and sample DNA source can severely affect
  %% estimates of de-novo deletion frequencies.
    
  %% Follow-up studies are needed to validate the findings.

\end{abstract}

\ifthenelse{\boolean{publ}}{\begin{multicols}{2}}{}

\section*{Background (from Manuscript \#1)}

Oral clefts are among the most common birth defects, and include three
anatomical defects: cleft lip (CL); cleft lip and palate (CLP) and
cleft palate (CP). Because there are similarities in embryological and
epidemiological evidence, CL and CLP are often grouped together as
cleft lip with/without cleft palate (CL/P), although there is debate
about whether all three groups may have distinct etiologies
\cite{Harville2005, Forrester2007}. Collectively, oral clefts
represent half of all craniofacial malformations and create a major
public health burden for both affected children and their
families. The overall prevalence of oral clefts is estimated at 1 per
700 live births worldwide, but there is dramatic variation across
populations and between racial and ethnic groups, in particular for
CL/P \cite{Mossey2002,Mossey2009b}.

Oral clefts show strong familial aggregation, and the recurrence risk
among first degree relatives is approximately 32 times greater than
the general population risk for CL/P, and approximately 56 times
greater for CP \cite{Sivertsen2008a}.  Twin studies also suggest a
major role for genes controlling risk to oral clefts with monozygotic
twins showing much higher concordance rates than dizygotic twins, 31\%
versus 2\% for CL/P, and 43\% versus 7\% for CP \cite{Mitchell2002}.
Normal development of craniofacial features is a complex process and
disruption of any one of numerous steps can lead to development of
oral clefts \cite{Stanier2004}, further supported by mounting evidence
that multiple genes, in addition to environmental influences, play a
role in the etiology of oral clefts \cite{Farrall1992,
  Schliekelman2002, Jugessur2009a, Beaty2010, Dixon2011, Ludwig2012}.

Assessment of chromosomal anomalies such as microdeletions and
translocations also played an important role in the identification of
genes and genomic regions underlying facial disorders
\cite{Bocian1987, Sander1994, Sander1995, Brewer1998, Brewer1999a,
  Schutte1999a, FitzPatrick2003, Alkuraya2006, Benko2009}.  In
particular, high throughput technologies such as CGH and SNP arrays
have gained popularity in assessing chromosomal alterations
\cite{Milunsky2008,Osoegawa2008}. For example, Sivertsen et.al
assessed the prevalence of duplications and deletions in the 22q11
region (DiGeorge syndrome region) in Norwegian offspring with open CP,
but did not detect an association \cite{Sivertsen2007}.  Shi et.~al
used SNP genotyping, DNA sequencing, high-resolution DNA microarray
analysis, and long-range PCR to characterize deletions in 333
candidate genes for orofacial clefting using 2,823 samples from 725
two and three-generation families ascertained through a proband having
CL/P \cite{Shi2009}.  The authors confirmed several de-novo deletions
in some of these candidate genes, in particular SUMO1, TBX1, and
TFAP2A, raising the possibility that genes or regulatory elements
contained within deleted regions might play a role in the etiology of
oral clefts. Further, high rates of Mendelian inconsistencies were
observed in 11 different genes, suggesting the existence of additional
micro-deletions.

%\setcounter{section}{1}
%\renewcommand{\thesection}{\arabic{section}}
\section*{Results and Discussion}
\emph{Lorem ipsum dolor sit amet, consectetur adipiscing elit.  Donec a diam lectus.  Sed sit amet ipsum mauris.  Maecenas congue ligula ac quam viverra nec consectetur ante hendrerit.  Donec et mollis dolor.  Praesent et diam eget libero egestas mattis sit amet vitae augue.  Nam tincidunt congue enim, ut porta lorem lacinia consectetur.  Donec ut libero sed arcu vehicula ultricies a non tortor.  Lorem ipsum dolor sit amet, consectetur adipiscing elit.  Aenean ut gravida lorem.  Ut turpis felis, pulvinar a semper sed, adipiscing id dolor.  Pellentesque auctor nisi id magna consequat sagittis.  Curabitur dapibus enim sit amet elit pharetra tincidunt feugiat nisl imperdiet.  Ut convallis libero in urna ultrices accumsan.  Donec sed odio eros.  Donec viverra mi quis quam pulvinar at malesuada arcu rhoncus.  Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.  In rutrum accumsan ultricies.  Mauris vitae nisi at sem facilisis semper ac in est.}

%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

\section*{Conclusions}
\emph{Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec a diam lectus. Sed sit amet ipsum mauris. Maecenas congue ligula ac quam viverra nec consectetur ante hendrerit. Donec et mollis dolor. Praesent et diam eget libero egestas mattis sit amet vitae augue. Nam tincidunt congue enim, ut porta lorem lacinia consectetur. Donec ut libero sed arcu vehicula ultricies a non tortor. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean ut gravida lorem. Ut turpis felis, pulvinar a semper sed, adipiscing id dolor. Pellentesque auctor nisi id magna consequat sagittis. Curabitur dapibus enim sit amet elit pharetra tincidunt feugiat nisl imperdiet. Ut convallis libero in urna ultrices accumsan. Donec sed odio eros. Donec viverra mi quis quam pulvinar at malesuada arcu rhoncus. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. In rutrum accumsan ultricies. Mauris vitae nisi at sem facilisis semper ac in est.}

%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
%\section*{R}
<<package, eval = TRUE, echo = FALSE>>=
library("Gviz")
library("trioClasses")
data("cnv", package = "trioClasses")
data("pedigrees", package="CleftCNVAssoc")
se <- SummarizedExperiment(assays = SimpleList(cnv = t(cnv.obj$cnv.mat)), colData = DataFrame(id=rownames(cnv.obj$cnv.mat), row.names = rownames(cnv.obj$cnv.mat)), rowData = cnv.obj$cmp.gr )
beaty.trios <- MinimumDistance:::trios(beaty.pedigree)
beaty.ped <- DataFrame(famid = do.call("rbind",strsplit(beaty.trios$O, "_" ))[,1], id = beaty.trios$O, fid = beaty.trios$F, mid = beaty.trios$M, sex = NA, dx = NA)
ped <- PedClass(beaty.ped)
fe <- FamilyExperiment(se, pedigree = ped )
@ 
<<FamilyExperiment, echo = FALSE, eval = TRUE>>=
fe.parents <- fe[,colnames(fe)%in%parents(fe)]
@ 
<<freq-vec, echo = FALSE, eval = TRUE>>=
freq.vec <- colSums(cnv(fe.parents))/nrow(cnv(fe.parents))
@ 
<<trioStates, eval = TRUE, echo = FALSE>>=
trioAssay <- trioClasses:::TrioAssay(fe, type = "cnv")
trioStates <- with(trioAssay, matrix( paste0(F,M,O), nrow = nrow(O), ncol = ncol(O)))
dimnames(trioStates) <- dimnames(trioAssay$O)
@ 
<<table-list, eval = TRUE, echo = FALSE>>=
table.list <- apply(trioStates, 2, "table")
@ 
<<transtab, eval = TRUE, echo = FALSE>>=
trans.vec <- as( lapply( table.list, trioClasses:::trans.tab ), "numeric")
#head(table.list[which(trans.vec <= 0.05/length(trans.vec))])
@ 

<<gviz, eval = TRUE, echo = FALSE>>=
#layout(matrix(1:4, nrow = 2, ncol = 2, byrow = TRUE ))
chr <- 8
gr.cnp <- cnv.obj$cmp.gr
gr.cnp.chr <- gr.cnp[as.logical(seqnames(gr.cnp) == paste0("chr", chr))]

p.vec <- trans.vec
p.vec.chr <- p.vec[as.logical(seqnames(gr.cnp) == paste0("chr", chr))]
freq.vec.chr <- freq.vec[as.logical(seqnames(gr.cnp) == paste0("chr", chr))]

#tr.rate.case.chr <- tr.rate.case[as.logical(seqnames(gr.cnp) == paste0("chr", chr))]


atrack <- AnnotationTrack(reduce(gr.cnp.chr), name = "CNP comp.", fill = "darkgreen")
gtrack <- GenomeAxisTrack()
dtrack <- DataTrack( range = gr.cnp.chr, data = -log10(p.vec.chr), type = "p", cex = 1, name = "-log10(p)", ylim = c(0,10), col = "red" )
dtrack2 <- DataTrack( range = gr.cnp.chr, data = freq.vec.chr, type = "s", col = "blue", cex = 1, name = "freq", ylim = c(0,1), lwd = 2 )
#dtrack2 <- DataTrack( range = gr.cnp.chr, data = tr.rate.case.chr, type = "p", cex = 1, name = "Cleft Trans. Rate", col = "red", type = "p", ylim = c(0,1) )
itrack <- IdeogramTrack(genome = "hg18", chromosome = paste0("chr", chr), lty = 1, lwd = 1 )
#data(geneModels)
#grtrack <- GeneRegionTrack(geneModels, genome="hg18", chromosome=chr, name="Gene Model")
#plotTracks(list(itrack, gtrack, atrack, ))
fud <- 1e5/2
@ 
<<n-tests, eval = TRUE, echo = FALSE>>=
n.tests <- sum(!is.na(trans.vec))
@ 
<<reduce-locus, eval = TRUE, echo = FALSE >>=
locus <- reduce(cnv.obj$cmp.gr[which(trans.vec <= 0.05/n.tests)])
wd <- width(locus)/1e3
@ 
<<trans.tab, eval = FALSE, echo = FALSE >>=
trioClasses:::trans.tab
@ 
\clearpage
%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
\section*{Methods}
\subsection*{Data Description}
\begin{itemize}
  \item Performed \Sexpr{n.tests} tests.  Bonferroni significant locus has width \Sexpr{wd} kB.
  \item \penn{} joint HMM
  \item european, MAD $<0.3$, non-WGA, aux $\ne 1$
  \item coverage $>10$
  \item \Sexpr{length(cnv.obj$gr)} hemi/homozygous deletions identified in \Sexpr{nrow(completeTrios(fe))} trios
  \item \Sexpr{ncol(cnv(fe))} CNV components
  \subitem Common ($>0.01$): \Sexpr{sum(freq.vec > 0.01)}
  \subitem Rare: \Sexpr{sum(freq.vec <= 0.01)}
  \item Construct trio-states for all CNV components
    \subitem recall that we use indicator variable for hemi/homozygous deletions
  \item must be at least 5 informative mating pairs
    \subitem 01x and 10x
  \item count transmissions and perform binom.test (See ``trans.tab'')
\end{itemize}

%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

In Figure \ref{gviz1} the \logp-values are displayed for the region on
chromosome 8 that showed genome-wide significance after Bonferroni
correction.  We see that the frequency of the CNV components
responsible for this signal have a frequency in the parents of nearly
0.5.

<<sig.table>>=
table.list[which(trans.vec <= 0.05/n.tests)][[1]]
@ 

\begin{figure}
  \begin{center}
<<gviz, fig = TRUE , echo = FALSE , eval = TRUE >>=
plotTracks(list(dtrack, dtrack2, gtrack, itrack ),  background.panel = "#FFFEDB", background.title = "darkblue", from = 39356825 - fud, to = 39497557 + fud)
@ 
\end{center}
\caption{$-\log p$-values for Fisher's exact test (blue) for an
  increase in transmission rate among cleft and control groups.  The
  transmission rate in the cleft group (red) is seen to increase
  around 20 MB with the $\log p$-value climbing to close to 3.}
\label{gviz1}
\end{figure}


%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

\begin{center} 
\begin{figure}
<<manhattan, fig = TRUE, echo = FALSE, eval = FALSE >>=
plot(-log10(trans.vec), pch = 20 )
@ 
\end{figure}
\end{center} 


%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


\begin{center}
\begin{figure}
<<freq-hist, echo = FALSE, eval = TRUE, fig = TRUE>>=
hist(freq.vec[freq.vec > 0.01], breaks = seq(0.01,1,by = 0.05), xlim = c(0.01,1), main = "Frequency of Common CNV Components", xlab = "frequency", ylab = "count")
@ 
\caption{histogram}
\end{figure}
\end{center}


%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

\clearpage

\section*{Authors contributions}
\section*{Acknowledgements}


{\ifthenelse{\boolean{publ}}{\footnotesize}{\small}
 \bibliographystyle{bmc_article} 
  \bibliography{sgy-master} }  

\ifthenelse{\boolean{publ}}{\end{multicols}}{}

\end{bmcformat}
\end{document}
