% Sweave(file = ``./deletion-tdt.Rnw'')
\documentclass[10pt, final]{article}
\usepackage{Sweave}
\SweaveOpts{eps=FALSE,echo=TRUE,figs.only=TRUE,keep.source=FALSE, prefix.string=figures/deletion-tdt}
%\usepackage{fullpage}
%\usepackage{times}
%\usepackage[colorlinks=TRUE,urlcolor=blue,citecolor=blue]{hyperref}
%\usepackage{ifthen}  % Conditional 
%\usepackage{multicol}   %Columns
\usepackage{graphicx} %should be removed before submission!
\input{/home/sgy/jhsph/latex/sgy}
\begin{document}
<<options, echo=FALSE, eval = TRUE, echo = FALSE>>=
  options(width=75, continue = " ")
  library("Gviz")
  library("trioClasses")
  library("TxDb.Hsapiens.UCSC.hg18.knownGene")

  data("pedigrees", package="CleftCNVAssoc")
  data("penncnvjoint", package = "CleftCNVAssoc")
@ 
<<se-beaty, echo = FALSE, eval = TRUE>>=
  data("cnv", package = "trioClasses")
  cnv.beaty.obj <- cnv.obj
  se.beaty <- SummarizedExperiment(assays = SimpleList(cnv = t(cnv.beaty.obj$cnv.mat)), colData = DataFrame(id=rownames(cnv.beaty.obj$cnv.mat), row.names = rownames(cnv.beaty.obj$cnv.mat)), rowData = cnv.beaty.obj$cmp.gr )
  beaty.trios <- MinimumDistance:::trios(beaty.pedigree)
  beaty.ped <- DataFrame(famid = do.call("rbind",strsplit(beaty.trios$O, "_" ))[,1], id = beaty.trios$O, fid = beaty.trios$F, mid = beaty.trios$M, sex = NA, dx = NA)
  ped <- PedClass(beaty.ped)
  fe <- FamilyExperiment(se.beaty, pedigree = ped )
@ 
<<se-pitt, echo = FALSE, eval = TRUE>>=
  data("cnv.pitt", package = "trioClasses")
  cnv.pitt.obj <- cnv.obj
  se.pitt <- SummarizedExperiment(assays = SimpleList(cnv = t(cnv.pitt.obj$cnv.mat)), colData = DataFrame(id=rownames(cnv.pitt.obj$cnv.mat), row.names = rownames(cnv.pitt.obj$cnv.mat)), rowData = cnv.pitt.obj$cmp.gr )
  pitt.trios <- MinimumDistance:::trios(pitt.pedigree)
  
pitt.ped <- DataFrame(famid = pitt.trios$O, id = pitt.trios$O, fid = pitt.trios$F, mid = pitt.trios$M, sex = NA, dx = NA)

  ped.pitt <- PedClass(pitt.ped)
  fe.pitt <- FamilyExperiment(se.pitt, pedigree = ped.pitt )
@ 

<<FamilyExperiment, echo = FALSE, eval = TRUE>>=
  fe.parents <- fe[,colnames(fe)%in%parents(fe)]
  fe.pitt.parents <- fe.pitt[,colnames(fe.pitt)%in%parents(fe.pitt)]
@ 
<<freq-vec, echo = FALSE, eval = TRUE>>=
    freq.vec <- colSums(cnv(fe.parents))/nrow(cnv(fe.parents))
    freq.pitt.vec <- colSums(cnv(fe.pitt.parents))/nrow(cnv(fe.pitt.parents))
@ 
<<trioStates, eval = TRUE, echo = FALSE>>=
    trioAssay <- trioClasses:::TrioAssay(fe, type = "cnv")
    trioStates <- with(trioAssay, matrix( paste0(F,M,O), nrow = nrow(O), ncol = ncol(O)))
    dimnames(trioStates) <- dimnames(trioAssay$O)
    trioAssay.pitt <- trioClasses:::TrioAssay(fe.pitt, type = "cnv")
    trioStates.pitt <- with(trioAssay.pitt, matrix( paste0(F,M,O), nrow = nrow(O), ncol = ncol(O)))
    dimnames(trioStates.pitt) <- dimnames(trioAssay.pitt$O)
@ 
<<table-list, eval = TRUE, echo = FALSE>>=
    table.list <- apply(trioStates, 2, "table")
    table.list.pitt <- apply(trioStates.pitt, 2, "table")
@ 
The top CNV component has the followinfg trio-states.  Where `1' indicates a deletion and order is F, M, O.
<<transtab, eval = TRUE, echo = FALSE>>=
    trans.vec <- as( lapply( table.list, trioClasses:::trans.tab ), "numeric")
    trans.rate <- as( lapply( table.list, trioClasses:::trans.rate ), "numeric")
    table.list[which(trans.vec <= 0.05/sum(!is.na(trans.vec)))][[1]]
    trans.vec.pitt <- as( lapply( table.list.pitt, trioClasses:::trans.tab ), "numeric")
    trans.rate.pitt <- as( lapply( table.list.pitt, trioClasses:::trans.rate ), "numeric")
@ 

\begin{center}
\begin{figure}
<<gviz, eval = TRUE, echo = FALSE, fig = TRUE, width = 8>>=
chr <- 8
gr.cnp <- cnv.beaty.obj$cmp.gr
gr.cnp.chr <- gr.cnp[as.logical(seqnames(gr.cnp) == paste0("chr", chr))]
    
p.vec <- trans.vec
p.vec.chr <- p.vec[as.logical(seqnames(gr.cnp) == paste0("chr", chr))]
freq.vec.chr <- freq.vec[as.logical(seqnames(gr.cnp) == paste0("chr", chr))]
trans.rate.chr <- trans.rate[as.logical(seqnames(gr.cnp) == paste0("chr", chr))]

TranscriptDb <- TxDb.Hsapiens.UCSC.hg18.knownGene
atrack <- AnnotationTrack(reduce(gr.cnp.chr), name = "CNP comp.", fill = "darkgreen")
gtrack <- GenomeAxisTrack()
dtrack <- DataTrack( range = gr.cnp.chr, data = -log10(p.vec.chr), type = "p", cex = 1, name = "-log10(p)", ylim = c(0,10), col = "red" )
dtrack2 <- DataTrack( range = gr.cnp.chr, data = ifelse(freq.vec.chr > 0.01, freq.vec.chr, NA ), type = "s", col = "blue", cex = 1, name = "Frequency", ylim = c(0,1), lwd = 2 )
dtrack3 <- DataTrack( range = gr.cnp.chr, data = trans.rate.chr, type = "s", col = "blue", cex = 1, name = "Transmission Rate", ylim = c(0,1), lwd = 2 )
itrack <- IdeogramTrack(genome = "hg18", chromosome = paste0("chr", chr), lty = 1, lwd = 1 )
grtrack <- GeneRegionTrack(TranscriptDb, genome="hg18", chromosome=chr, name="UCSC Known Transcripts", showId = TRUE)

fud <- 5e5/2
plotTracks(list(dtrack, dtrack2, dtrack3, grtrack, gtrack, itrack ),  background.panel = '#FFFEDB', background.title = 'darkblue', from = 39356825 - fud, to = 39497557 + fud)
@ 
\label{chr8-locus}
\caption{Cleft Trios --- Red: $-\log_{10}p$-values for each CNV component that has at least \textbf{five} ``0/1'' mating pairs.  Null hypothesis is transmission rate of $\frac{1}{2}$, and alternative hypothesis is ``one-sided.''  Blue: The frequency of CNV component in the parents.  Yellow: Gene tracks, labeled by transcript.} 
\end{figure}
\end{center}

\begin{center}
\begin{figure}
<<gviz2, eval = TRUE, echo = FALSE, fig = TRUE, width = 8>>=
chr <- 8
gr.cnp <- cnv.pitt.obj$cmp.gr
gr.cnp.chr <- gr.cnp[as.logical(seqnames(gr.cnp) == paste0("chr", chr))]
    
p.vec <- trans.vec.pitt
p.vec.chr <- p.vec[as.logical(seqnames(gr.cnp) == paste0("chr", chr))]
freq.vec.chr <- freq.pitt.vec[as.logical(seqnames(gr.cnp) == paste0("chr", chr))]
trans.rate.pitt.chr <- trans.rate.pitt[as.logical(seqnames(gr.cnp) == paste0("chr", chr))]

TranscriptDb <- TxDb.Hsapiens.UCSC.hg18.knownGene
atrack <- AnnotationTrack(reduce(gr.cnp.chr), name = "CNP comp.", fill = "darkgreen")
gtrack <- GenomeAxisTrack()
dtrack <- DataTrack( range = gr.cnp.chr, data = -log10(p.vec.chr), type = "p", cex = 1, name = "-log10(p)", ylim = c(0,10), col = "red" )
dtrack2 <- DataTrack( range = gr.cnp.chr, data = ifelse(freq.vec.chr > 0.01, freq.vec.chr, NA ), type = "s", col = "blue", cex = 1, name = "Frequency", ylim = c(0,1), lwd = 2 )
dtrack3 <- DataTrack( range = gr.cnp.chr, data = trans.rate.pitt.chr, type = "s", col = "blue", cex = 1, name = "Transmission Rate", ylim = c(0,1), lwd = 2 )
itrack <- IdeogramTrack(genome = "hg18", chromosome = paste0("chr", chr), lty = 1, lwd = 1 )
grtrack <- GeneRegionTrack(TranscriptDb, genome="hg18", chromosome=chr, name="UCSC Known Transcripts", showId = TRUE)

fud <- 5e5/2
plotTracks(list(dtrack, dtrack2, dtrack3, grtrack, gtrack, itrack ),  background.panel = '#FFFEDB', background.title = 'darkblue', from = 39356825 - fud, to = 39497557 + fud)
@ 
\label{chr8-locus-con}
\caption{Control Trios --- Red: $-\log_{10}p$-values for each CNV component that has at least \textbf{five} ``0/1'' mating pairs.  Null hypothesis is transmission rate of $\frac{1}{2}$, and alternative hypothesis is ``one-sided.''  Blue: The frequency of CNV component in the parents.  Yellow: Gene tracks, labeled by transcript.} 
\end{figure}
\end{center}
\begin{center}
  \begin{figure}
    \includegraphics[width = 4in]{./figures/ucsc}
    \caption{UCSC Genome Browser at chromosome 8 locus.}
  \end{figure}
\end{center}
<<n-tests, eval = TRUE, echo = FALSE>>=
    n.tests <- sum(!is.na(trans.vec))
@ 
The CNV components with signiifcant $p$-values (Bonferroni) are given below.
<<reduce-locus, eval = TRUE, echo = FALSE >>=
    (locus <- cnv.beaty.obj$cmp.gr[which(trans.vec <= 0.05/n.tests)])
    wd <- width(reduce(locus))/1e3
@ 
All CNV components are contiguous and the total width is \Sexpr{round(wd,1)} kB.
<<trans.tab, eval = TRUE, echo = FALSE >>=
    trioClasses:::trans.tab
@ 
<<pitt, eval = TRUE, echo = TRUE >>=
gr.deletion.pitt <- gr.pitt[values(gr.pitt)$numsnp >= 10 & values(gr.pitt)$cn %in% 0:1 ]
sum(countOverlaps( gr.deletion.pitt, reduce(locus) ))
@ 
\begin{center}
\begin{figure}
<<hist, fig = TRUE, echo = FALSE, eval = TRUE, width = 8>>=
par(bg = '#FFFEDB')
layout(matrix(1:2, nrow = 1, ncol = 2))
hist(trans.rate, breaks = 20, xlim = c(0,1), ylim = c(0,200), main = "Cleft Trios", xlab = "Transmission Rate", col='darkblue')
hist(trans.rate.pitt, breaks = 20, xlim = c(0,1), ylim = c(0,200), main = "Control Trios", xlab = "Transmission Rate", col='darkblue')
@ 
\end{figure}
\end{center}
\section*{Methods}
\subsection*{Cleft Data Description}
\begin{itemize}
  \item Performed \Sexpr{n.tests} tests.  Bonferroni significant locus has width \Sexpr{wd} kB.
  \item \penn{} joint HMM
  \item european, MAD $<0.3$, non-WGA, aux $\ne 1$
  \item coverage $>10$
  \item \Sexpr{length(cnv.obj$gr)} hemi/homozygous deletions identified in \Sexpr{nrow(completeTrios(fe))} trios
  \item \Sexpr{ncol(cnv(fe))} CNV components
  \subitem Common ($>0.01$): \Sexpr{sum(freq.vec > 0.01)}
  \subitem Rare: \Sexpr{sum(freq.vec <= 0.01)}
  \item Construct trio-states for all CNV components
    \subitem recall that we use indicator variable for hemi/homozygous deletions
  \item must be at least 5 informative mating pairs
    \subitem 01x and 10x
  \item count transmissions and perform binom.test (See ``trans.tab'')
\end{itemize}
<<transmission-rate>>=
mean(trans.rate[subjectHits(findOverlaps(reduce(locus), cnv.beaty.obj$cmp.gr))])
mean(trans.rate.pitt[subjectHits(findOverlaps(reduce(locus), cnv.pitt.obj$cmp.gr))])
@  
<<tdt-p, echo = TRUE, eval = TRUE>>=
mean(-log10(trans.vec[subjectHits(findOverlaps(reduce(locus), cnv.beaty.obj$cmp.gr))]))
mean(-log10(trans.vec.pitt[subjectHits(findOverlaps(reduce(locus), cnv.pitt.obj$cmp.gr))]))
@  
<<frequencies, echo = TRUE, eval = TRUE>>=
freq.pitt.vec[subjectHits(findOverlaps(reduce(locus), cnv.pitt.obj$cmp.gr))]
freq.vec[subjectHits(findOverlaps(reduce(locus), cnv.beaty.obj$cmp.gr))]
@ 
\end{document}
