% Sweave(file = "./trioClasses.Rnw")
\documentclass[10pt]{article}

\usepackage{fullpage}
\usepackage{times}
\usepackage[colorlinks=TRUE,urlcolor=blue,citecolor=blue]{hyperref}
%\title{TITLE}
%\author{Sam}
%\date{\today}
%\input{/home/bst/student/syounkin/jhsph/latex/sgy}
%\input{/home/sgy/jhsph/latex/sgy}
\usepackage{Sweave}
\begin{document}
\setlength{\parskip}{0.2\baselineskip}
\setlength{\parindent}{0pt}
%\maketitle
Today is \today.
%<<cachedir, echo=FALSE>>=
%setCacheDir( path = './cache/' )
%@
\begin{Schunk}
\begin{Sinput}
> rm(list = ls())
> source("~/jhsph/R/reload.R")
> library("trioClasses")
> library("trio")
\end{Sinput}
\end{Schunk}
\begin{Schunk}
\begin{Sinput}
> data(ped)
> pedigreeInfo <- within(ped.df, {
     F <- as.character(fid)
     M <- as.character(mid)
     O <- as.character(id)
 })
> pedigreeInfo <- subset(pedigreeInfo, !is.na(F) & !is.na(M) & 
     !is.na(O))
> tg.ped <- Pedigree(pedigreeInfo = pedigreeInfo)
> tg.ped
\end{Sinput}
\begin{Soutput}
This pedigree object contains 1812 complete trios.
For access to the data frame use the trios() accessor function.
\end{Soutput}
\end{Schunk}
\begin{Schunk}
\begin{Sinput}
> id <- offspringNames(tg.ped)
> fid <- fatherNames(tg.ped)
> mid <- motherNames(tg.ped)
> names(fid) <- names(mid) <- id
> c(length(unique(id)), length(unique(fid)), length(unique(mid)))
\end{Sinput}
\begin{Soutput}
[1] 1812 1812 1812
\end{Soutput}
\end{Schunk}
\section{TrioSet}
\begin{Schunk}
\begin{Sinput}
> data(geno)
> head(geno.mat[, 1:6])
\end{Sinput}
\begin{Soutput}
        snp1 snp2 snp3 snp4 snp5 snp6
578_01     2    2    0    2    2    1
578_02     1    0    1    0    2    2
578_03     0    2    2    0    2    2
1539_01    2    0    2    2    2    0
1539_02    1    1    2    0    1    0
1539_03    1    0    2    2    0    2
\end{Soutput}
\begin{Sinput}
> geno.trio <- genoMat(tg.ped, geno.mat)
> (tg.ped.comp <- completeTrios(tg.ped, colnames(geno.trio)))
\end{Sinput}
\begin{Soutput}
This pedigree object contains 33 complete trios.
For access to the data frame use the trios() accessor function.
\end{Soutput}
\begin{Sinput}
> (ts <- TrioSet(tg.ped.comp, geno = geno.trio))
\end{Sinput}
\begin{Soutput}
TrioSet (storageMode: lockedEnvironment)
assayData: 10 features, 33 samples 
  element names: geno 
protocolData: none
phenoData: none
featureData
  featureNames: snp1 snp2 ... snp10 (10 total)
  fvarLabels: position chromosome isSnp
  fvarMetadata: labelDescription
experimentData: use 'experimentData(object)'
Annotation:  
genome:  hg19 
\end{Soutput}
\end{Schunk}
\begin{Schunk}
\begin{Sinput}
> geno <- getGeno(ts)
> (aTDT <- allelicTDT(mat.snp = t(geno), size = 10000))
\end{Sinput}
\begin{Soutput}
      Allelic TDT

Top 5 SNPs:
      Statistic p-value
snp10    3.8571 0.04953
snp4     2.5714 0.10881
snp5     1.1429 0.28505
snp8     1.0000 0.31731
snp6     0.6923 0.40538
\end{Soutput}
\end{Schunk}
\end{document}
