% Sweave(file = "./trioClasses.Rnw")
\documentclass[10pt]{article}
<<echo=FALSE>>=
  options(width=70, continue = " ")
@
\SweaveOpts{eps=FALSE,echo=TRUE,figs.only=TRUE,keep.source=FALSE, prefix.string=figures/trioClasses}
\usepackage{fullpage}
\usepackage{times}
\usepackage[colorlinks=TRUE,urlcolor=blue,citecolor=blue]{hyperref}
%\title{TITLE}
%\author{Sam}
%\date{\today}
%\input{/home/bst/student/syounkin/jhsph/latex/sgy}
%\input{/home/sgy/jhsph/latex/sgy}
\begin{document}
\setlength{\parskip}{0.2\baselineskip}
\setlength{\parindent}{0pt}
%\maketitle
Today is \today.
%<<cachedir, echo=FALSE>>=
%setCacheDir( path = './cache/' )
%@
<<packages, cache = FALSE>>=
rm(list=ls())
source("~/jhsph/R/reload.R")
library("trioClasses")
library("trio")
@
<<sample-ped>>=
data(ped)
head(ped.df)
pedigreeInfo <- within(ped.df, {
  F <- as.character(fid)
  M <- as.character(mid)
  O <- as.character(id)
})
tg.ped <- Pedigree(pedigreeInfo = pedigreeInfo)
tg.ped
@
<<ids, echo = FALSE, eval = FALSE>>=
id <- offspringNames(tg.ped)
fid <- fatherNames(tg.ped)
mid <- motherNames(tg.ped)
names(fid) <- names(mid) <- id
c(length(unique(id)),length(unique(fid)),length(unique(mid)))
@
\section{TrioSet}
<<trioset>>=
data(geno)
head(geno.mat[,1:6])
geno.trio <- genoMat(tg.ped, geno.mat )
( tg.ped.comp <- completeTrios(tg.ped, colnames(geno.trio)) )
( object.gTrio <- gTrio( tg.ped.comp, geno = geno.trio ) )
@ 
<<geno>>=
geno <- getGeno(object.gTrio, type = "holger")
( aTDT <- allelicTDT( mat.snp = t(geno), size = 1e4 ) )
@ 
\end{document}
