% Sweave(file = "./trioClasses.Rnw")
\documentclass[10pt]{article}
<<echo=FALSE>>=
  options(width=70, continue = " ")
@
\SweaveOpts{eps=FALSE,echo=TRUE,figs.only=TRUE,keep.source=FALSE, prefix.string=figures/trioClasses}
\usepackage{fullpage}
\usepackage{times}
\usepackage[colorlinks=TRUE,urlcolor=blue,citecolor=blue]{hyperref}
\title{The R package \emph{trioClasses} for definition of the class \emph{SNPTrioExperiment}, an extension of \emph{SummarizedExperiment}, for use in trio based analyses of genetic data.}
\author{Samuel G. Younkin, Robert Scharpf, Holger Schwender, Ingo Ruczinski}
\date{\today}
%\input{/home/bst/student/syounkin/jhsph/latex/sgy}
%\input{/home/sgy/jhsph/latex/sgy}
\begin{document}
\setlength{\parskip}{0.2\baselineskip}
\setlength{\parindent}{0pt}
\maketitle
\section{Packages}
<<package>>=
library("trioClasses")
@
\section{Sample Data}
<<data, echo = TRUE>>=
data("sample")
se <- SummarizedExperiment(assays = SimpleList(geno = t(sm)), colData = col.DF, rowData = gr )
@ 
\subsection{Pedigree}
<<ped>>=
ped <- PedClass(ped.DF)
@ 
\subsection{SNPTrioExperiment}
<<ste>>=
ste <- SNPTrioExperiment(se, pedigree = ped )
@ 
<<ste-show>>=
ste
@ 
\section{Methods}
\subsection{Holger-style Genotype Matrix}
Coercion to matrix for \emph{trio}
<<trio>>=
geno <- as(ste, "matrix")
aTDT(geno[,1:5])
@ 
Or apply the aTDT method to the SNPTrioExperiment directly.
<<trio2>>=
aTDT(ste[1:5])
@ 
\subsection{Count of Transmission of Rare Variants (\emph{TransCount()})}
First we define the world and a block.
<<block>>=
world <- range(gr)
block <- GRanges( seqnames = "chr14", IRanges(start(gr[100]), width = 10e3), strand = "*")
@ 
\subsubsection{In a block}
<<transcount>>=
block
TransCount( ste[MAF(ste) <= 0.01,], block )
@ 
\subsubsection{Not In the block}
<<transcount>>=
setdiff(world,block)
TransCount( ste[MAF(ste) <= 0.01,], setdiff(world,block) )
@ 
\subsubsection{In the ``world''}
<<transcount>>=
world
TransCount( ste[MAF(ste) <= 0.01,], world )
@ 
\subsection{Iteratively}
<<iter>>=
trans.vec <- TransCount( ste[MAF(ste) <= 0.01,], gr[1:10] + 1e3 )
@ 
<<transvec>>=
table(trans.vec)
@ 
\clearpage
%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
\section{Classes}
\subsection{SnpMatrix, DataFrame, GRanges} 
The four key ingredients are the SNP matrix, the pedigree information as a DataFrame, position of the SNPs given by a GRanges object, and covariate data given as a DataFrame.
<<grdf2>>=
sm
ped.DF
gr
@ 
\subsection{SummarizedExperiment}
We combine three of the key ingredients when we create the SummarizedExperiment object.
<<se>>=
se <- SummarizedExperiment(assays = SimpleList(geno = t(sm)), colData = col.DF, rowData = gr )
@ 
\subsection{SNPTrioExperiment}
Now, we include the pedigree information as an object of class PedClass.  We keep PedClass independent of SNPTrioExperiment for flexibility.
<<ste>>=
ped <- PedClass(ped.DF)
ste <- SNPTrioExperiment(se, pedigree = ped )
@ 
Here is the show method.
<<ste-show>>=
ste
@ 
And now we verify that it is indeed an extension of SummarizedExperiment.
<<ste-class>>=
getClass("SNPTrioExperiment")
@ 
\subsection{PedClass}
Now we investigate the pedigree slot of the SNPTrioExperiment object.
<<classes>>=
class(pedigree(ste))
getClass("PedClass")
pedigree(ste)
@
\subsection{geno accessor}
<<geno>>=
class(geno(ste))
getClass("SnpMatrix")
geno(ste)
@ 
\subsection{RowData}
<<rowdata>>=
rowData(ste)
@ 
\subsection{ColData}
<<coldata, eval = FALSE, echo = FALSE>>=
colData(ste)
@ 
\section{Validity}
\end{document}
